cdap-master/src/main/java/co/cask/cdap/data/tools/flow/FlowQueuePendingCorrector.java

```diff
@@ -147,60 +151,60 @@ public void run() throws Exception {
     System.out.println("Running queue.pending correction");
     List<NamespaceMeta> namespaceMetas = namespaceQueryAdmin.list();
     for (NamespaceMeta namespaceMeta : namespaceMetas) {
-      run(Id.Namespace.from(namespaceMeta.getName()));
+      run(new NamespaceId(namespaceMeta.getName()));
     }
   }
 
   /**
    * Corrects queue.pending metric for all flowlets in a namespace.
    */
-  public void run(Id.Namespace namespaceId) throws Exception {
+  public void run(NamespaceId namespaceId) throws Exception {
     System.out.println("Running queue.pending correction on namespace " + namespaceId);
     Collection<ApplicationSpecification> apps = store.getAllApplications(namespaceId);
     for (ApplicationSpecification app : apps) {
-      Id.Application appId = Id.Application.from(namespaceId, app.getName());
+      ApplicationId appId = new ApplicationId(namespaceId.getNamespace(), app.getName());
       run(appId, app);
     }
   }
 
   /**
    * Corrects queue.pending metric for all flowlets in an application.
    */
-  public void run(Id.Application appId) throws Exception {
+  public void run(ApplicationId appId) throws Exception {
     ApplicationSpecification app = store.getApplication(appId);
     run(appId, app);
   }
 
   /**
    * Corrects queue.pending metric for all flowlets in an application.
    */
-  public void run(Id.Application appId, ApplicationSpecification appSpec) throws Exception {
+  public void run(ApplicationId appId, ApplicationSpecification appSpec) throws Exception {
     System.out.println("Running queue.pending correction on app " + appId);
-    Preconditions.checkArgument(appSpec.getName().equals(appId.getId()),
+    Preconditions.checkArgument(appSpec.getName().equals(appId.getApplication()),
                                 String.format("Expected appSpec name '%s' to be equal to appId name '%s'",
-                                              appSpec.getName(), appId.getId()));
+                                              appSpec.getName(), appId.getApplication()));
     for (FlowSpecification flow : appSpec.getFlows().values()) {
-      run(Id.Flow.from(appId, flow.getName()));
+      run(new ProgramId(appId, ProgramType.FLOW, flow.getName()));
     }
   }
 
   /**
    * Corrects queue.pending metric for a flow.
    */
-  public void run(final Id.Flow flowId) throws Exception {
-    ApplicationSpecification app = store.getApplication(flowId.getApplication());
+  public void run(final ProgramId flowId) throws Exception {
```

Comment:
for now, in these ``run`` methods, it is better to add ``Preconditions.checkArgument(ProgramType.FLOW == flowId.getProgramType())`` as the first step, until we have a ``FlowId`` class.